### DISCUSSION TITLE
Shopware Snippets (i18n) in Frontends

### DISCUSSION THREAD

**Discussion by BrocksiNet** [shopware/frontends#132] [Ideas]
_**Original Question from [Tim Resl](https://shopwarecommunity.slack.com/team/U01Q319HD4N):**_

Hey, i have a question regarding i18n or Shopware Snippets. 

According to a conversation with [@Björn](https://shopwarecommunity.slack.com/team/U015WB0KL2Y) there is a Pull Request on an i18n integration. https://github.com/shopware/frontends/pull/124 I think that makes sense. It is fast and super for npm but not flexible for shop owners.

The question for me is: why we do not take Shopware Snippets instead of json files? You can provide them via the store API. You could limit the snippets by namespace.

**What do you think about?**

**Comment by BrocksiNet** [comment]
Here is the summary of what we have discussed about the topic in our daily work.

**Possible Process:**
- From an API, we need to get the snippets (store or admin).
- During build time, we can generate a json file from these snippets.
- This json file can then be used for translation in the frontend store.

**Issues:**

- The client has to wait for the next build to see their changes.
- We write a json file with some snippets, but we don't know if the snippets will be used somewhere in the frontend store.
- We need to filter the snippets otherwise maybe the file gets too big (admin snippets).
- This makes the build process more complex.

**Conclusion:**
For now, we think that static but specific files are sufficient for translating the static template content. Also, we assume that these translations will not change that often and should be a minimum. Products, categories and so on will be translated if the right context is set. Also feel free to integrate this function into your project if the need is there from the customer.

**Reply by timresl** [reply]
Thanks for the quick answer @BrocksiNet

**Comment by mkucmus** [comment]
little update: related issue #504 now has a PR #596
